### ISSUE TITLE
Start of the code review process 👋

### ISSUE THREAD

**Issue by longnd** [hsnbd/data-scraping#23]
Hello Baker👋 , 
Thank you for your effort on the code submission. I am Long, engineering lead at Nimble, and I am happy to be the reviewer for our code review session.

During the review process, I would like to know more about your decisions, and thus I will create issues where I think there could be more improvements regarding your performance here.

At the same time, please keep in mind that this is a bi-directional process, and I would love to hear back from you as well. Therefore, do not hesitate to raise your questions or share your opinions about the implementation (if any) during the process.

If we are aligned on any issue, and you would like to correct them, please address the issue using a proper git flow (creating a new branch, opening a Pull Request (PR) per issue, and merging the code when you are ready), and I will follow up on those fixes. Please note that you don't have to close any of my created issues after merging your PRs, I will help verify and close them for you once they pass. 😇

We expect the code review process to be completed within 2-3 days at most. As a result, ensure you are responsive during this process. If you need additional time, please inform us right away so that we can plan accordingly.

In the end, I really do hope that you find the process enjoyable. Good luck and happy coding. 🤘

**Comment by hsnbd** [comment]
Please confirm if I should use axios instead of puppeteer. Because the #27 #24 #25 issues depend on it.

**Comment by hsnbd** [comment]
If I add a new issue by myself would it be okay to fix the problem and merge PRs?

**Comment by longnd** [comment]
hi @hsnbd I'm sorry for the late response. 

> If I add a new issue by myself, would it be okay to fix the problem and merge PRs?

I'm not sure what issue you would like to add but I assume you have identified some issues and would like to work on fixing them, that would be greatly appreciated. 
However, as mentioned before, I wanted to focus on addressing the most critical concerns during this code review process to ensure that we can complete it within a few days. The feedback and changes we have discussed so far are sufficient for me to assess the code.

I have responded to all the open issues and also reviewed the PR where you implemented the unit tests. It is now time to conclude this stage of the code review process. Thank you for your efforts thus far. Our Talent Acquisition team will provide feedback to you via email.
